telemetry_package/main.py

```diff
@@ -1,28 +1,74 @@
-import sys
+import json
 import rclpy
-from rclpy.node import Node
+import socket
+import os
+import select
 
+from rcl_interfaces.msg import Log
 from common_package_py.common_node import CommonNode
 
+DEFAULT_UNIX_SOCKET_PATH = "/tmp/thi_drone"
 
 class TelemetryNode(CommonNode):
-    """A class that represents a telemetry node.
-    """
 
-    def __init__(self, id: str):
-        """Creates a new TelemetryNode.
+    def __init__(self):
+        # The creation of the needed sockets takes place in the constructor
+        super().__init__('telemetry_node')
 
-        Args:
-            id (str): Unique node id
-        """
+        if os.path.exists(DEFAULT_UNIX_SOCKET_PATH): 
+            print('Socket already exists')
+            raise FileExistsError
+            
+        # Create server socket
+        self.server_sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
 
-        super().__init__(id)
+        try:
+            # Bind server socket and listen for a connection. The node is blocked as long as a connection does not exist
+            self.server_sock.bind(DEFAULT_UNIX_SOCKET_PATH)
+            self.server_sock.listen(1)
+
+            # Getting the reference for the client socket
+            readable, _, _, = select.select([self.server_sock], [], [])
+            for sock in readable:
+                if sock == self.server_sock:
+                    # Accept the connection
+                    self.client_sock, _ = self.server_sock.accept()
+                    print('connection accepted. Creating Subscription')
+
+                    '''
+                    Creating the subscription after the connection has accepted ensures that the callback function is executed only after the soccket is connected
+                    '''
+                    self.subscription = self.create_subscription(
+                        Log,
+                        '/rosout',
+                        self.subscription_callback, 
+                        100)
```

Comment:
You have to explicitly set the Quality of Service here.
100 means that ROS makes sure that every single message will reach it's destination (with a cache of 100 messages; TCP like).
This is not what we want: We want the telemetry node to receive the debug output in a UDP like style.
To achieve this, you have to set the Quality of Service accordingly.
You can either consult the docs or ask Johan on what you have to do here.